WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCoupon.swift

```diff
@@ -243,7 +246,8 @@ struct AddEditCoupon: View {
             }
             .sheet(isPresented: $showingSelectCategories) {
                 ProductCategorySelector(isPresented: $showingSelectCategories,
-                                        config: ProductCategorySelector.Configuration.categoriesForCoupons,
+                                        viewConfig: categorySelectorConfig,
+                                        categoryListConfig: categoryListConfig,
```

Comment:
I would like to understand the reason for injecting the view model and configurations separately.

What do you think about moving these configurations to `ProductCategorySelectorViewModel`?

Reply:
My reasoning is that these configs are solely for the UI layer, so I didn't want to move them to the view model and then have to access the view model to get the configs. Does that make sense?